storage/innobase/log/log0recv.cc

```diff
@@ -3811,6 +3811,8 @@ void recv_sys_t::apply(bool last_batch)
 
   mysql_mutex_lock(&mutex);
 
+  if (last_batch)
+    dblwr.pages.clear();
```

Comment:
I thought that we could do this inside the existing `if (last_batch)` earlier in this function. There is no need to introduce one more condition. Already at that point we should have finished applying log records (and reading pages).